### ISSUE TITLE
Learn gcn from directed graph

### ISSUE THREAD

**Issue by hyungsseop** [rusty1s/pytorch_geometric#2915]
Hi, I am learning directed graph through gcn.

In another question, I found in the directed graph that gcn learning is not a problem.

What I am curious about is the process of building the normalized Laplacian matrix of directed graph.

In the gcn paper, sysmmetric laplacian matrix is created using an undirected graph. but The Laplacian matrix in the directed graph seems non-sysmmetric.

I wonder if there is any problem with this difference in gcn training.

And if you know, please let me know if you have any additional comments on the effect.

thank you.

**Comment by rusty1s** [comment]
Do you desperately need symmetric normalization? Other-wise, using standard `mean` aggregation like in `SAGEConv` is more straightforward to apply.

**Comment by hyungsseop** [comment]
It's not necessary. I want to know what is different between directed and undirected graphs.

**Comment by rusty1s** [comment]
There is no difference during message passing. All message passing layers will send their messages along the given edges in the graph, and this works fine both for undirected and directed graphs. However, in general, for directed graphs, you also wanna make sure that messages are passed in reverse direction, i.e.:
```python
out = conv1(x, edge_index) + conv2(x, edge_index[torch.tensor([0, 1])]
```

**Comment by dhorka** [comment]
> However, in general, for directed graphs, you also wanna make sure that messages are passed in reverse direction as well (using a different set of trainable parameters)
I'm curious about that, is there any paper or someplace where I can find an explanation for this? I mean why do you need to pass the message in reverse direction too?

**Comment by rusty1s** [comment]
I think this is just a trick that most papers apply silently. It's originally proposed in [this paper](https://arxiv.org/abs/1703.06103) I believe. From my point of view, this is crucial to help the GNN in training, as it now can also gather information of the reverse connection, e.g., the information from out-going edges.

**Comment by dhorka** [comment]
But this equivalent to just convert the directed graph to an undirected one, isn't it?

**Comment by rusty1s** [comment]
No, since you learn a different set of parameters for reverse message passing.
